### PR TITLE
Added checks for reservations

### PR DESCRIPTION
**Pull request by jaytaph**

### CODE REVIEW

File: internal/reservation.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"net/http"
+
+	"github.com/bitmaelum/bitmaelum-suite/pkg/bmcrypto"
+	"github.com/bitmaelum/bitmaelum-suite/pkg/hash"
+)
+
+// @TODO: Make this dynamic
+const baseReservedUrl = "https://resolver.bitmaelum.org/reserved/"
+
+var SkipReservationCheck = false
```
Comment (jaytaph) [inline internal/reservation.go:37]: I'm not 100% happy with this, but I'm not sure how to make sure that the unit-tests will pass without these checks (as they will fail during unit-testing)

File: internal/reservation.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/bitmaelum/bitmaelum-suite/pkg/bmcrypto"
+	"github.com/bitmaelum/bitmaelum-suite/pkg/hash"
+)
+
+// @TODO: Make this dynamic
+const baseReservedUrl = "https://resolver.bitmaelum.org/reserved/"
```
Comment (jaytaph) [inline internal/reservation.go:35]: maybe we should make this a repo instead, and have a mock / real implementation that we can switch during testing?

Comment (acalatrava) [reply internal/reservation.go:35]: yeah, that would be nice

**Review by acalatrava — APPROVED** [review]
LGTM

**Pull request merged**